objects/src/chain/mod.rs

```diff
@@ -0,0 +1,49 @@
+use super::{AdviceInputsBuilder, Felt, Mmr, ToAdviceInputs, ZERO};
+
+// TODO: Consider using a PartialMmr that only contains the Mmr nodes that are relevant to the
+// transaction being processed.
```

Comment:
Yeah - I think we'll need to do this in the future as loading full chain MMR into the advice provider would be too much overhead. The process could work as follows:

On the client side (i.e., user wallet):
- We'd only store partial MMR to begin with - so, no extra work needed.

On the node side:
- We take the set of input notes, and based on these extract a partial MMR from the full MMR (which will likely be stored in memory).
- Then, construct this `ChainMmr` from the partial MMR built in step 1.

Alternatively, we could build advice inputs directly from the full MMR - but the interface would need to be slightly different.